tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -1225,8 +1227,47 @@ Status Converter::BuildCudaEngine(
   }
 
 #if IS_TRT_VERSION_GE(8, 0, 0, 0)
-  builder_config->setFlag(nvinfer1::BuilderFlag::kSPARSE_WEIGHTS);
-  VLOG(1) << "Setting sparsity for TensorRT8!";
+  enum class SparseComputeMode {
+    DISABLED,
+    ENABLED,
+    BENCHMARK
+  };
+
+  static SparseComputeMode sparse_compute_mode = [](bool _enable_sparse_compute)
+  {
+    SparseComputeMode _sparse_compute_mode;
+    int64 _sparse_benchmark_mode;
+    /*This feature is only to be used for benchmarking purpose. This feature
+    artificially introduce a sparse weight pattern compatible with Sparse
+    TensorCores of NVIDIA Ampere GPU architecture. It will complete corrupt
+    the numerical values of the computation, it shall only be used to evaluate
+    the benefit of using sparse computation for inference*/
+    TF_CHECK_OK(tensorflow::ReadInt64FromEnvVar("TF_TRT_SPARSE_BENCHMARK_MODE",
+                                                /*default_val=*/-1,
+                                                &_sparse_benchmark_mode));
+
+    string sparse_log_msg = "[TF-TRT] Sparse compute capability is ";
+    if (_sparse_benchmark_mode >= 0) {
+      sparse_log_msg = StrCat(sparse_log_msg, "set to benchmark mode. ",
+          "Shall only be used for sparse computing benchmark and debug.");
+      _sparse_compute_mode = SparseComputeMode::BENCHMARK;
+    }
+    else if (_enable_sparse_compute) {
+      sparse_log_msg = StrCat(sparse_log_msg, "enabled.");
+      _sparse_compute_mode = SparseComputeMode::ENABLED;
+    } else {
+      sparse_log_msg = StrCat(sparse_log_msg, "disabled.");
+      _sparse_compute_mode = SparseComputeMode::DISABLED;
+    }
+    LOG(INFO) << sparse_log_msg;
+
+    return _sparse_compute_mode;
+  }(enable_sparse_compute);
```

Comment:
Can we extend the env variable to avoid the need of API change here?

Reply:
You prefer to entirely rely on an ENV Var ? No Python API ? I thought Env var flags were supposed to be exclusively debug features and temporary features. This one is here to stay

Reply:
We discussed this at the meeting: to just enable the feature and no need to an API change to enable it, and the env is for debugging purpose.
However, you mentioned that you prefer an API change to enable it. If that is the case, please split this into two PR, to facilitate the reviewing for the API change, as we discussed in the meeting.

Reply:
Waiting for author to respond to this comment.

Reply:
@DEKHTIARJonathan Any update on this PR? Please. Thank you!

Reply:
@DEKHTIARJonathan Any update on this PR? Please. Thank you!

Reply:
@DEKHTIARJonathan Any update on this PR? Please. Thank you!

Reply:
@DEKHTIARJonathan Any update on this PR? Please. Thank you!

Reply:
@DEKHTIARJonathan Any update on this PR? Please. Thank you!

Reply:
@DEKHTIARJonathan Any update on this PR? Please. Thank you!

Reply:
Will update very soon. Sorry for the delay

Reply:
Solved